lib/reek/smells/smell_detector.rb

```diff
@@ -24,32 +25,9 @@ class SmellDetector
       # in any configuration file.
       DEFAULT_EXCLUDE_SET = []
 
-      class << self
-        def contexts
-          [:def, :defs]
-        end
-
-        # :reek:UtilityFunction
-        def default_config
-          {
-            SmellConfiguration::ENABLED_KEY => true,
-            EXCLUDE_KEY => DEFAULT_EXCLUDE_SET.dup
-          }
-        end
-
-        def inherited(subclass)
-          subclasses << subclass
-        end
-
-        def descendants
-          subclasses
-        end
-
-        private
-
-        def subclasses
```

Comment:
Hard to see in the diff, but I got completely rid of `subclasses`. Now we just use `descendants` everywhere.